kernel-rs/src/sysfile.rs

```diff
@@ -84,7 +85,7 @@ pub unsafe fn sys_fstat() -> usize {
     let (_, f) = ok_or!(argfd(0), return usize::MAX);
     // user pointer to struct stat
     let st = ok_or!(argaddr(1), return usize::MAX);
-    (*f).stat(st) as usize
+    (*f).stat(st).map_or_else(|_| usize::MAX, |_| 0)
```

Comment:
- Closure 안에 있는 게 cost 가 있는 연산이 아니므로, `map_or` 를 쓰는 게 더 깔끔하게 보입니다.
 - 그런데 `map_or` 는 error 가 첫 번째 인자로 오네요.. 위의 문장과 순서가 달라 실수하기 쉬우므로, 다음처럼 하는 게 좋겠습니다:
```suggestion
    ok_or!((*f).stat(st), return usize::MAX);
    0
```

Reply:
https://github.com/kaist-cp/rv6/pull/211/commits/ad6dc6bc031da3c43079506f566b1bfe0c38ee5a